neo/SmartContract/InteropService.cs

```diff
@@ -546,15 +546,26 @@ private static bool PutEx(ApplicationEngine engine, StorageContext context, byte
             if (value.Length > MaxStorageValueSize) return false;
             if (context.IsReadOnly) return false;
             if (!CheckStorageContext(engine, context)) return false;
+
             StorageKey skey = new StorageKey
             {
                 ScriptHash = context.ScriptHash,
                 Key = key
             };
-            StorageItem item = engine.Snapshot.Storages.GetAndChange(skey, () => new StorageItem());
-            if (item.IsConstant) return false;
-            item.Value = value;
-            item.IsConstant = flags.HasFlag(StorageFlags.Constant);
+
+            if (engine.Snapshot.Storages.TryGet(skey)?.IsConstant == true) return false;
```

Comment:
If we could access here to TrackableItem, we can save the `GetAndChange`

Reply:
`GetAndChange` here doesn't access the internal storage. It read from the cache because we call `TryGet` before.

Reply:
But if we can know if the item has `trackable.State = TrackState.Added;` we can avoid the two reads

Reply:
If it is `TrackState.Added`, both the `TryGet` and `GetAndChange` calls read from the cache. These are all automatic.